distributed/scheduler.py

```diff
@@ -261,6 +263,7 @@ def __init__(
         self.worker_restrictions = dict()
         self.resource_restrictions = dict()
         self.loose_restrictions = set()
+        self.retries = defaultdict(lambda: 0)
```

Comment:
It might be nice to keep this as a normal dict.  The common case is likely to be no retries at all.  It would be nice not to keep the extra `{key: 0}` around in that case.